Helper/FilterMapper.php

```diff
@@ -0,0 +1,149 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Cmp\Helper;
+
+use Magento\Catalog\Model\Layer\Filter\Item;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Store\Model\Store;
+use Nosto\Operation\Recommendation\Filters;
+use Nosto\Tagging\Helper\Data as NostoHelperData;
+
+class FilterMapper
+{
+    /** @var Filters */
+    private $filters;
+
+    /** @var NostoHelperData */
+    private $nostoHelperData;
+
+    /** @var string */
+    private $brand;
+
+    /**
+     * FilterMapper constructor.
+     * @param Filters $filters
+     * @param NostoHelperData $nostoHelperData
+     */
+    public function __construct(Filters $filters, NostoHelperData $nostoHelperData)
+    {
+        $this->filters = $filters;
+        $this->nostoHelperData = $nostoHelperData;
+    }
+
+    /**
+     * @param Store $store
+     */
+    public function init(Store $store): void
+    {
+        $this->brand = $this->nostoHelperData->getBrandAttribute($store);
+    }
+
+    /**
+     * @param Item $item
+     * @throws LocalizedException
+     */
+    public function mapFilter(Item $item): void
+    {
+        /** @var string $frontendInput */
+        $frontendInput = $item->getFilter()->getData('attribute_model')
+            ->getData('frontend_input');
+
+        $filterName = $item->getName();
+        $value = '';
+        switch ($frontendInput) {
+            case 'price':
+                $value = $item->getData('value');
+                break;
+            case 'select':
+                $value = $item->getData('label');
+                break;
+            case 'multiselect':
+                $value = $item->getData('label');
+                break;
```

Comment:
It can be merged with the parent case. Like:
```suggestion
case 'select':
case 'multiselect':
    $value = $item->getData('label');
    break;
```